backend/src/main/java/de/backend/service/SpeciesService.java

```diff
@@ -0,0 +1,50 @@
+package de.backend.service;
+
+import de.backend.model.Species;
+import de.backend.repo.SpeciesRepo;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class SpeciesService {
+
+    private final SpeciesRepo speciesRepo;
+
+    public final IdService idService;
+
+    public SpeciesService(SpeciesRepo speciesRepo, IdService idService) {
+        this.speciesRepo = speciesRepo;
+        this.idService = idService;
+    }
+
+    public List<Species> list() {
+        return speciesRepo.findAll();
+    }
+    public Species findById(String id) {
+        Optional<Species> optionalSpecies = speciesRepo.findById(id);
+        if (optionalSpecies.isPresent()) {
+            return optionalSpecies.get();
+        }
+        throw new IllegalArgumentException("Id not found!");
```

Comment:
`speciesRepo.findById(id).orElseThrow(() -> new IllegalArgumentException("Id not found!"))`